pkg/healthcheck/healthcheck.go

```diff
@@ -2264,34 +2263,152 @@ func (hc *HealthChecker) checkMisconfiguredOpaquePortAnnotations(ctx context.Con
 	return nil
 }
 
-func misconfiguredOpaquePortAnnotationsInService(service *corev1.Service, pods []*corev1.Pod) error {
-	for _, pod := range pods {
-		if err := misconfiguredOpaqueAnnotation(service, pod); err != nil {
-			return err
+// addEndpointAddressPods primarily takes an endpoint address, and if it's
+// target is a pod that is not already in pods, then it adds it.
+func addEndpointAddressPods(pods []*corev1.Pod, addr corev1.EndpointAddress, kubeAPI *controllerK8s.API, service *corev1.Service) ([]*corev1.Pod, error) {
```

Comment:
We don't need to pass in the service as an argument, right?  Just the namespace?